src/main/java/com/gradle/enterprise/conventions/customvalueprovider/GradleEnterpriseConventions.java

```diff
@@ -42,7 +41,7 @@
 
     public GradleEnterpriseConventions(ProviderFactory providerFactory) {
         this.providerFactory = providerFactory;
-        this.gradleEnterpriseServerUrl = getSystemProperty(GRADLE_ENTERPRISE_URL_PROPERTY_NAME, PUBLIC_GRADLE_ENTERPRISE_SERVER, providerFactory);
```

Comment:
That now forces every employee at Gradle to configure the Gradle enterprise URL, right? I think that isn't good. It would be awesome if we could detect whether the current user is authenticated against ge.gradle.org, and if they aren't, then try to publish to `scans.gradle.org` if a scan is requested?

Or maybe it is simpler to configure a new system property which actively selects `scans.gradle.com` (aka the free public instance).

Reply:
> It would be awesome if we could detect whether the current user is authenticated against ge.gradle.org, and if they aren't, then try to publish to scans.gradle.org if a scan is requested?

It would be, but seems not possible now: https://gradle.slack.com/archives/C032LFSKY/p1608018204417500

> Or maybe it is simpler to configure a new system property which actively selects scans.gradle.com (aka the free public instance).

I assume you mean to make `scans.gradle.com` the default value, it seems not work as well (Gradle complains "you're not authenticated with server scans.gradle.com"), see the discussion above.